spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1466,245 +1470,216 @@ def handle_max_vel(self, request: SetVelocity.Request, response: SetVelocity.Res
-                    == full_body_feedback.feedback.payload_estimation_feedback.status.STATUS_SMALL_MASS
-                ):
-                    return GoalResponse.SUCCESS
-                if (
-                    full_body_feedback.feedback.payload_estimation_feedback.status.value
-                    == full_body_feedback.feedback.payload_estimation_feedback.status.STATUS_ERROR
-                ):
-                    return GoalResponse.FAILED
                 return GoalResponse.IN_PROGRESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_CONSTRAINED_MANIPULATION_FEEDBACK_SET
+            if (
+                feedback.command.claw_gripper_feedback.status.value
+                == feedback.command.claw_gripper_feedback.status.STATUS_UNKNOWN
             ):
-                if (
-                    full_body_feedback.feedback.constrained_manipulation_feedback.status.value
-                    == full_body_feedback.feedback.constrained_manipulation_feedback.status.STATUS_RUNNING
-                ):
-                    return GoalResponse.IN_PROGRESS
-                return GoalResponse.FAILED
-            else:
+                self.get_logger().error("ERROR: claw grippper status unknown")
                 return GoalResponse.IN_PROGRESS
+            # else: STATUS_AT_GOAL or STATUS_APPLYING_FORCE
+        else:
+            self.get_logger().error("ERROR: unknown gripper command type")
+            return GoalResponse.IN_PROGRESS
+        return None
 
-        elif feedback.command.command_choice == feedback.command.COMMAND_SYNCHRONIZED_FEEDBACK_SET:
+    def _robot_command_goal_complete(self, feedback: RobotCommandFeedback) -> GoalResponse:
+        if feedback is None:
+            # NOTE: it takes an iteration for the feedback to get set.
+            return GoalResponse.IN_PROGRESS
+
+        choice = feedback.command.command_choice
+        if choice == feedback.command.COMMAND_FULL_BODY_FEEDBACK_SET:
+            full_body_feedback = feedback.command.full_body_feedback
+            return self._process_full_body_command_feedback(full_body_feedback)
+
+        elif choice == feedback.command.COMMAND_SYNCHRONIZED_FEEDBACK_SET:
             # The idea here is that a synchronized command can have arm, mobility, and/or gripper
             # sub-commands in it.  The total command isn't done until all sub-commands are satisfied.
             # So if any one of the sub-commands is still in progress, it short-circuits out as
             # IN_PROGRESS.  And if it makes it to the bottom of the function, then all components
             # must be satisfied, and it returns SUCCESS.
-            # One corner case to know about is that the commands that don't provide feedback, such
+            # One corner if choice == to know about is that the commands that don't provide feedback, such
```

Comment:
I think this may have been an unintentional find/replace change?

Reply:
Definitely. Thank you for catching this

Reply:
@abaker-bdai